src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -2104,3 +2106,123 @@ def set_attention_slice(self, slice_size: Optional[int]):
 
         for module in modules:
             module.set_attention_slice(slice_size)
+
+
+class EfficiencyMixin:
```

Comment:
Can we maybe rename this:
```suggestion
class StableDiffusionMixin:
```

Reply:
But seems like it doesn't just affect the Stable Diffusion family, though. 

If a similar mixin class were to be designed what would be the process? It might be better to have a base `EfficiencyMixin` class and then use it as a subclass to write more pipeline-specific classes such as `StableDiffusionEfficiencyMixin`. 

I don't like the `StableDiffusionMixin` name -- it's uninformative and confusing in light of `DiffusionPipeline`.